inc/switchtec/switchtec.h

```diff
@@ -241,6 +241,38 @@ struct switchtec_fw_image_info {
 	unsigned long secure_version;
 };
 
+struct switchtec_fw_metadata_gen4 {
```

Comment:
Changes to the library should be in another patch.

This is a low level hardware interface that should not be exposed outside of the library.

If we need to print more information we should extend the existing interfaces, not bypass them.

Reply:
Done. Implemented the library changes as a different commit. I have put the metadata definition back to the library and added new variables to the existing structure to expose the metadata fields from the flash partition. 
I had to expose the switchtec_part_info() as this provides a interface to get the flash info for a particular partition instead of all the flash partitions that the current exposed function provides.